x/evm/vm/interface.go

```diff
@@ -0,0 +1,66 @@
+package vm
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+)
+
+// PrecompiledContracts defines a map of address -> precompiled contract
+type PrecompiledContracts map[common.Address]vm.PrecompiledContract
+
+type StatefulPrecompiledContract interface {
+	vm.PrecompiledContract
+	RunStateful(evm EVM, addr common.Address, input []byte, value *big.Int) (ret []byte, err error)
+}
+
+// EVM defines the interface for the Ethereum Virtual Machine used by the EVM module.
+type EVM interface {
```

Comment:
@fedekunze how did you arrive at this list? E.g. `(evm *EVM) Config` is not implemented on geth

Reply:
we need to access the public fields from the EVM as well